.github/workflows/ci-build.yml

```diff
@@ -1,27 +1,73 @@
-name: Build and Tests
+name: 'Build and Test' 
 
 on:
   push:
-    branches: [ main ]
   pull_request:
     branches: [ main ]
 
 jobs:
-  build:
+  Build:
+
+    runs-on: ubuntu-latest
+    env: 
+      VSTEST_CONNECTION_TIMEOUT: 900
 
+    strategy:
+      matrix:
+        dotnet-version: [ 'netcoreapp3.1', 'net5', 'net6' ] 
```

Comment:
We should build against `netcoreapp3.1 netstandard2.0 net461 net6`

Reply:
Will rewrite.